custom_components/localtuya/number.py

```diff
@@ -102,7 +104,8 @@ def device_class(self):
 
     async def async_set_native_value(self, value: float) -> None:
         """Update the current value."""
-        await self._device.set_dp(value, self._dp_id)
+        send_value = int(value) if self._config.get(CONF_VALUE_TYPE) == "int" else value
```

Comment:
Does this need to be a bit more detailed? Ie it does int casting; should it also do string/float/bool/etc?

Reply:
In my testing with a lot of tuya devices, I have found only float and int values are used for numbers.